### DISCUSSION TITLE
Disable for html tags

### DISCUSSION THREAD

**Discussion by mvpopuk** [HiPhish/rainbow-delimiters.nvim#78] [Q&A]
Hey friends, great plugin. I was curious if I could disable it for HTML or HTML tags. Thanks

**Comment by HiPhish** [comment]
Have you tried blacklisting HTML? See `:h g:rainbow_delimiters`

**Comment by mannas006** [comment]
In Neovim, you can control the behavior of plugins, including whether they are enabled or disabled for specific filetypes or file patterns. If you want to disable a plugin for HTML files or HTML tags, you can use the `ftplugin` or `autocmd` (autocommand) functionality in your Neovim configuration.

Assuming you're using a plugin manager like vim-plug or dein.vim, and the plugin you want to disable is already installed, you can use the following steps:

1. Locate the plugin configuration section in your `init.vim` or `init.lua` file, where you have configured the plugin you want to disable.

2. Add a conditional statement to disable the plugin for HTML files or tags. For example, if you're using vim-plug, your configuration might look like this:

  ```
 ```vim
   " vim-plug section
   call plug#begin('~/.vim/plugged')

   " Your other plugins...

   " Plugin you want to disable for HTML
   if !exists('g:loaded_matchtag')  " Replace 'matchtag' with the actual plugin name
     Plug 'username/pluginname'
   endif

   " Your other plugins...

   call plug#end()
   ```

3. Add an autocmd or ftplugin to disable the plugin for HTML files. For example:

   ```vim
   " Disable the plugin for HTML files
   autocmd FileType html let g:loaded_matchtag = 1
   ```

   Replace `html` with the actual filetype you want to disable the plugin for.

4. Save your configuration file and restart Neovim.

This example assumes that the plugin uses a global variable like `g:loaded_matchtag` to determine whether it should be loaded. Please replace `matchtag` with the actual variable used by the plugin you want to disable.

If the plugin doesn't provide a convenient global variable, you might need to explore the plugin's documentation or source code to identify the appropriate mechanism for disabling it selectively.

**Reply by HiPhish** [reply]
Literally none of what you wrote applies to this plugin. Did you query a language model and just copy-paste the result here without verification? Please do not post this garbage here, it is spam that only confuses users who have legitimate questions and are looking for help. Consider this a warning.

**Comment by HiPhish** [comment]
For anyone who finds this thread in the future, you can blacklist a language inside the `g:rainbow_delimiters` variable:

```vim
" Vim script
let g:rainbow_delimiters = {'blacklist': ['html']}
```

```lua
-- Lua
vim.g.rainbow_delimiters = {blacklist = {'html'}
```

As mentioned above, see `:h g:rainbow_delimiters` for all the possible keys and values for that variable.

**Comment by Dillpickleschmidt** [comment]
For anyone in the future trying to use this without JSX tag coloring, here's my setup mimicking VSCode:
```lua
return {
  "HiPhish/rainbow-delimiters.nvim",
  event = "BufReadPre",
  config = function()
    -- Set up highlight groups
    vim.api.nvim_set_hl(0, "RainbowDelimiterYellow", { fg = "#FFD700" })
    vim.api.nvim_set_hl(0, "RainbowDelimiterViolet", { fg = "#AE5FAE" })
    vim.api.nvim_set_hl(0, "RainbowDelimiterBlue", { fg = "#179FFF" })

    local rainbow_delimiters = require 'rainbow-delimiters'

    ---@type rainbow_delimiters.config
    vim.g.rainbow_delimiters = {
      query = {
        [''] = 'rainbow-delimiters',
        -- Use rainbow-parens to exclude JSX tags
        tsx = 'rainbow-parens',
        jsx = 'rainbow-parens',
      },
      highlight = {
        'RainbowDelimiterYellow',
        'RainbowDelimiterViolet',
        'RainbowDelimiterBlue',
        'RainbowDelimiterYellow',
        'RainbowDelimiterViolet',
        'RainbowDelimiterBlue',
        'RainbowDelimiterYellow',
      },
    }
  end,
}
```
If you're curious, inspect the 3 files in `ls ~/.local/share/nvim/lazy/rainbow-delimiters.nvim/queries/tsx` to see why I used 'rainbow-parens'.
